src/main/java/nextstep/subway/section/domain/Sections.java

```diff
@@ -159,27 +159,68 @@ private boolean possibleAddedLastSection(Section targetSection) {
     }
 
     public void deleteSection(Station station) {
-        if (!getStations().contains(station)) {
-            throw new NotFoundStationException();
-        }
-        if (isNotLastStation(station)) {
-            throw new IsNotLastStationException();
+        validateDeleteSection(station);
+
+        if (possibleDeletedFirstSection(station)) {
+            deleteFirstSection();
+            return;
         }
-        if (size() == 1) {
-            throw new DeleteSectionException();
+        if (possibleDeletedLastSection(station)) {
+            deleteLastSection();
+            return;
         }
+        deleteMiddleSection(station);
+    }
 
-        Section lastSection = getLastSection();
+    private boolean possibleDeletedFirstSection(Station station) {
+        return getFirstSection().getUpStation().equals(station);
```

Comment:
```suggestion
return getFirstSection().isUpStation(station);
```

만들어둔 메소드를 활용할 수 있을것 같아요 😄